contentcuration/contentcuration/frontend/shared/utils/validation.js

```diff
@@ -0,0 +1,397 @@
+  if (!mastery || mastery.type !== MasteryModelsNames.M_OF_N) {
+    return [];
+  }
+  return getMasteryModelNValidators()
+    .map(validator => validator(mastery.n))
+    .filter(value => value !== true);
+}
+
+/**
+ * @param  {Object} nodeDetails Node data like title, license, ...
+ * @param  {Array} assessmentItems An array of the node's assessment items
+ *                                 (for exercise nodes)
+ * @return {Boolean} Returns `true` if node details (and assessment items for exercises) are valid.
+ *                   Returns `false` otherwise.
+ */
+export function isNodeComplete(nodeDetails, assessmentItems = []) {
+  if (validateNodeDetails(nodeDetails).length) {
+    return false;
+  }
+
+  if (nodeDetails.kind === ContentKindsNames.EXERCISE) {
+    if (!assessmentItems || !assessmentItems.length) {
+      return false;
+    }
+
+    const isInvalid = assessmentItem => {
+      const sanitizedAssessmentItem = sanitizeAssessmentItem(assessmentItem, true);
+      return validateAssessmentItem(sanitizedAssessmentItem).length;
+    };
+    if (assessmentItems.some(isInvalid)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * Validate node details - title, license etc.
+ * @param {Object} node A node.
+ * @returns {Array} An array of error codes.
+ */
+export function validateNodeDetails(node) {
+  let errors = [];
+
+  const titleErrors = validateNodeTitle(node);
+  if (titleErrors.length) {
+    errors = errors.concat(titleErrors);
+  }
+
+  // authoring information is required for resources
+  if (!node.freeze_authoring_data && node.kind !== ContentKindsNames.TOPIC) {
+    const licenseErrors = validateNodeLicense(node);
+    const copyrightHolderErrors = validateNodeCopyrightHolder(node);
+    const licenseDescriptionErrors = validateNodeLicenseDescription(node);
+
+    if (licenseErrors.length) {
+      errors = errors.concat(licenseErrors);
+    }
+    if (copyrightHolderErrors.length) {
+      errors = errors.concat(copyrightHolderErrors);
+    }
+    if (licenseDescriptionErrors.length) {
+      errors = errors.concat(licenseDescriptionErrors);
+    }
+  }
+
+  // mastery is required on exercises
+  if (node.kind === ContentKindsNames.EXERCISE) {
+    const masteryModelErrors = validateNodeMasteryModel(node);
+    const masteryModelMErrors = validateNodeMasteryModelM(node);
+    const masteryModelNErrors = validateNodeMasteryModelN(node);
+
+    if (masteryModelErrors.length) {
+      errors = errors.concat(masteryModelErrors);
+    }
+    if (masteryModelMErrors.length) {
+      errors = errors.concat(masteryModelMErrors);
+    }
+    if (masteryModelNErrors.length) {
+      errors = errors.concat(masteryModelNErrors);
+    }
+  }
+
+  return errors;
+}
+
+/**
+ * Validate node files - correct types, no associated errors, etc.
+ * @param {Array} files An array of files for a node.
+ * @returns {Array} An array of error codes.
+ */
+export function validateNodeFiles(files) {
```

Comment:
Could you add in my changes from this PR as well? https://github.com/learningequality/studio/pull/2278/commits/2275e9f83809b6f15c6a9196f40910952c84ac63#diff-b5d58672729f225f3203f8295b54e92aR135

We should allow topics and exercises not to have primary files